Highlight/Highlighter.php

```diff
@@ -563,12 +563,49 @@ public function highlightAuto($code, $languageSubset = null)
      * setAutodetectLanguages will turn on autodetection for all supported
      * languages.
      *
+     * @param bool $include_aliase Specify whether language aliases
+     *      should be included as well.
+     *
      * @return array
      *      An array of language names (strings).
      */
-    public function listLanguages()
+    public function listLanguages($include_aliases = false)
     {
+        if ($include_aliases === true ) {
+            return array_merge(self::$languages, array_keys(self::$aliases));
+        }
+
         return self::$languages;
     }
 
+    /**
+     * Returns list of all available aliases for given language name.
+     *
+     * @param string $language Name or alias of language to look-up.
+     *
+     * @return array
+     *      An array of all aliases associated with the requested
+     *      language name language. Passed-in name is included as
+     *      well.
+     */
+    public function getAliasesForLanguage($language)
+    {
+        if (!in_array($language, self::$languages)) {
+            if (array_key_exists($language, self::$aliases)) {
```

Comment:
Can you use `isset()` instead? `self::$aliases` should never contain `null` as a value but just to be safe.